Add iks-life-preserver terraform code (#94)
IBM-Cloud/kube-samples: iks-life-preserver/README.md

# iks-life-preserver

## What is this?

> Life Preserver is a simple project designed to be used with the 
> IBM Cloud Schematics service to make the process of inviting 
> someone to your IBM Cloud account for the purpose of helping you 
> troubleshoot issues with your IBM Cloud Kubernetes Service Clusters
>  (including OpenShift on IBM Cloud clusters).

Reviewer: ```suggestion
The Life Preserver project uses IBM Cloud Schematics to 
quickly invite a member of IBM Cloud Support to your IBM 
Cloud account. With this temporary access, IBM Cloud Support 
can help you troubleshoot issues with your IBM Cloud Kubernetes 
Service and Red Hat OpenShift on IBM Cloud clusters.
```

>  (including OpenShift on IBM Cloud clusters).

Author: @Rachael-Graham once merged, I'm totally fine with inclusion in the IBM Cloud docs site.

## How it works
Follow the steps below to invite a user to your account and grant them: 

* A `Viewer` IAM platform policy that allows the user to see only Kubernetes and OpenShift clusters
* A `Reader` IAM service policy that allows the user to have **view-only** access to the **specific cluster you grant them access to**. 

> Note: In terms of Kubernetes RBAC, this user will not have access to any Secrets or RBAC resources inside the cluster.

### Step 1

 Login to the IBM Cloud web console at https://cloud.ibm.com.

### Step 2 

> Browse to the IBM Cloud Schematics offering at https://cloud.ibm.com/schematics

Reviewer: ```suggestion
[Open the IBM Cloud Schematics dashboard.](https://cloud.ibm.com/schematics)
```

### Step 3

> Click `Create workspace` and give your new workspace a relevant name and pick a location and click `Create`. 

Reviewer: ```suggestion
Click **Create Workspace**. Enter a name, select a location, and click **Create**.
```

Reviewer: does this workspace need to be in the same RG as the cluster?

Author: It does not.

### Step 4

> Now you should be looking at your workspace settings. Provide `https://github.com/jpapejr/iks-life-preserver/tree/main` as the Terraform template URL and ensure that you choose a Terraform version of 0.12 or higher for this workspace. Click `Save template information` to save your changes. Wait for the system import and process the Terraform data before continuing.

Reviewer: ```suggestion
The workspace settings pane opens. In the **Import your Terraform template** section, enter `https://github.com/BM-Cloud/kube-samples/iks-life-preserver/tree/main` for the Terraform template **URL**, and select a **Terraform version** of `terraform_v0.12` or later.

Click **Save template information** and wait for Schematics to import and process the Terraform data before you continue.
```

### Step 5

> You should now see four variables that need to be populated before continuing:

Reviewer: ```suggestion
In the **Variables** section, enter **Override values** for the following variables:
```

> * `reference` - This is a user-defined reference that will be appeanded to the generated service ID name for purpose reference.
> * `cluster` - This is the **cluster ID** of the specific cluster to grant **read-only** access to.
> * `apikey` - This is an IBM Cloud API Key with sufficient privileges to invite new members to the account. For security purposes, ensure you check the `Sensitive` box on the far right so your key is obscured from view after it input.
> * `account` - This is the target account within which the operations will occur. 

Reviewer: moved some around to match the order in the ui
```suggestion
* `cluster` - The ID of the cluster to grant _read-only_ access to.
* `apikey` - An IBM Cloud API key with [sufficient privileges to invite users to the account](https://cloud.ibm.com/docs/account?topic=account-iamuserinv#invite-access). For security purposes, select the **Sensitive** checkbox to obscure your API key from view after changes are saved.
* `reference` - A reference that is appended to the generated service ID name.
* `account` - The IBM Cloud account ID which the cluster is created in.
```

> When done, click `Save changes`.

Reviewer: ```suggestion
Click `Save changes`.
```

### Step 6

> At the top of the screen click on the `Generate plan` button. Ensure it completes successfully before moving on. Address any Terraform errors that might arise due to back API Key or cluster ID values. 

Reviewer: ```suggestion
At the top of the settings pane, click **Generate plan**. In the **Activity** pane, monitor events to ensure that the plan generates successfully before you continue. To review any Terraform errors, click **View log** on the `Failed to generate plan` event card. Then, change the variable override values in the settings pane, and generate the plan again.
```

### Step 7

> Again, at the top of the screen, click on the `Apply plan` button. Once the `apply` operation has completed, you can click on `View Logs` to find the generated Service ID API Key. Share this securely with the individual providing assistance; they will be able to authenticate using this API Key for the targeted account & cluster resources as defined above. 

Reviewer: ```suggestion
After the plan generates successfully, click **Apply plan**. After the plan applies successfully, click **View Log** and find the generated Service ID API key. Securely share this generated API key with the IBM Cloud Support engineer who is providing assistance. IBM Cloud Support uses this API key to authenticate with the specified account and view the cluster. 
```

### Step 8

> At this point, if the `apply` operation was successful you should be able to browse into the [IAM portion of the console](https://cloud.ibm.com/iam/users) to confirm the access. 

Reviewer: ```suggestion
Optional: In the [Users dashboard of the IAM console](https://cloud.ibm.com/iam/users), confirm the access of the IBM Cloud Support engineer. 
```

> ### Step 9 (roll-back/revert the process)

Reviewer: ```suggestion
### Step 9 (Revoke access)
```

> From the main workspace page, click on the `Actions` drop-down button near the top-right part of the screen and select `Delete`:

Reviewer: In the overview for the Schematics workspace, click the **Actions...** drop-down list and select `Delete`.

Author: updated in https://github.com/IBM-Cloud/kube-samples/pull/94/commits/62fa18b4f63dbce7796907aa3d4a193d97776fa9

> * If you want to remove all the changes and delete the workspace completely, check both boxes.

Reviewer: ```suggestion
* To remove all changes and delete the workspace completely, select both checkboxes.
```

> * If you want to remove all the changes and keep the workspace around for re-use at a later time, just check the second box. 

Reviewer: ```suggestion
* To remove all changes and keep the workspace for re-use at a later time, select only the second checkbox. 
```

> Confirm the operation by typing in the workspace name and then clicking `Delete`

Reviewer: ```suggestion
Confirm the operation by entering the workspace name and clicking **Delete**.
```
